Source/Core/AbsyExpr.cs

```diff
@@ -3837,6 +3843,206 @@ public T Dispatch<T>(IAppliableVisitor<T> visitor)
     }
   }
 
+  public class FieldAccess : IAppliable
+  {
+    private IToken _tok;
+
+    private string fieldName;
+    
+    private DatatypeSelector selector;
+
+    public IToken tok
+    {
+      get
+      {
+        Contract.Ensures(Contract.Result<IToken>() != null);
```

Comment:
Are these contracts useful? They don't run at compile-time. I would prefer not using them and then this `tok` becomes a regular auto-property `public IToken tok {get;set;}`

Reply:
They are all over the place in Boogie code. I was trying to be consistent. 

@RustanLeino : Could you shed some light? Should we try to drop these contracts gradually?

Reply:
Yes, unfortunately, these contracts are no longer useful, since the tool that would do the bytecode rewriting for them is no longer available. `Contract.Assert` and `Contract.Assume` are the only ones that (in a DEBUG build) generate run-time checks.

I think that dropping them gradually is a good plan. Those that remain in the source code still serve as documentation, until they will eventually be removed.

Reply:
I have removed most of the contracts in the classes I added. I also converted to properties wherever I could.